may-2018-reenabled-opcodes.md

```diff
@@ -0,0 +1,428 @@
+languages (with Python being a notable exception) differ from the strict mathematical definition.  Script follows the C language set of
+rules.  Namely:
+1. Non-integer quotients are rounded towards zero
+2. The equation `(a/b)*b + a%b == a` is satisfied by the results
+3. From the above equation it follows that: `a%b == a - (a/b)*b`
+4. In practice if `a` is negative for the modulo operator the result will be negative or zero.
+
+
+### OP_DIV
+
+    Opcode (decimal): 150
+    Opcode (hex): 0x96
+    
+Return the integer quotient of `a` and `b`.  If the result would be a non-integer it is rounded *towards* zero.
+
+    a b OP_DIV -> out
+    
+    where a and b are interpreted as numeric values
+    
+The operator must fail if:
+1. `!isnum(a) || !isnum(b)`. Fail if either operand is not a numeric value.
+1. `b == 0`. Fail if `b` is equal to any type of zero.
+
+Impact of successful execution:
+* stack memory use reduced
+* number of elements on stack is reduced by one
+
+Unit tests:
+1. `a b OP_DIV -> failure` where `!isnum(a)` or `!isnum(b)`. Both operands must be numeric values
+2. `a 0 OP_DIV -> failure`. Division by positive zero (all sizes), negative zero (all sizes), `OP_0` 
+4. `27 7 OP_DIV -> 3`, `27 -7 OP_DIV -> -3`, `-27 7 OP_DIV -> -3`, `-27 -7 OP_DIV -> 3`. Check negative operands. 
+  *Pay attention to sign*.
+5. check valid results for operands of different lengths `0..4`
+    
+### OP_MOD
+
+    Opcode (decimal): 151
+    Opcode (hex): 0x97
+
+Returns the remainder after dividing a by b.  The output will be represented using the least number of bytes required. 
+
+	a b OP_MOD -> out
+	
+	where a and b are interpreted as numeric values
+	
+The operator must fail if:
+1. `!isnum(a) || !isnum(b)`. Fail if either operand is not a numeric value.
+1. `b == 0`. Fail if `b` is equal to any type of zero.
+
+Impact of successful execution:
+* stack memory use reduced (one element removed)
+* number of elements on stack is reduced by one
+
+Unit tests:
+1. `a b OP_MOD -> failure` where `!isnum(a)` or `!isnum(b)`. Both operands must be numeric values.
+2. `a 0 OP_MOD -> failure`. Division by positive zero (all sizes), negative zero (all sizes), `OP_0` 
+4. `27 7 OP_MOD -> 6`, `27 -7 OP_MOD -> 6`, `-27 7 OP_MOD -> -6`, `-27 -7 OP_MOD -> -6`. Check negative operands. 
+  *Pay attention to sign*.
+5. check valid results for operands of different lengths `0..4`
+
+## New operations
+
+### OP_BIN2NUM
+
+*`OP_BIN2NUM` replaces `OP_LEFT` and uses it's opcode*
+
+    Opcode (decimal): 128
+    Opcode (hex): 0x80
+
+Convert the byte sequence into a numeric value, including minimal encoding.
+
+    `x1 OP_BIN2NUM -> n`
+
+See also `OP_NUM2BIN`.
+
+Notes:
+* if `x1` is any form of zero, including negative zero, then `OP_0` must be the result
+    
+Examples:
+* `0x0000000002 OP_BIN2NUM -> 0x02`
+* `0x800005 OP_BIN2NUM -> 0x85`
+
+The operator must fail if:
```

Comment:
There is no section here for "Impact of successful execution" as with the other opcodes.

Reply:
thanks, added.